### PR TITLE
Pore constraint

### PR DESCRIPTION
**Pull request by fweik**
Description of changes:
 - New SimplePore shape + tests
 - additional operators for Vector<>
 - Pointers to member function as setters/getters in ScriptInterface::AutoParameter
 - Python-style print function

Simpler, faster and more correct than Pore.

PR Checklist
------------
 - [x] Tests?
   - [x] Interface
   - [x] Core 
 - [x] Docs?

### CODE REVIEW

**Comment by fweik** [comment]
@KaiSzuttor  can you please have a look.

**Comment by fweik** [comment]
@KaiSzuttor

File: src/core/shapes/SimplePore.cpp [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    auto const dz = c_z - z;
+
+    /* Rescale to surface */
+    auto const d = std::sqrt(dr * dr + dz * dz);
+    auto const fac = (d - m_smoothing_rad) / d;
+
+    return {fac * dr, fac * dz};
+  }
+}
+
```
Comment (KaiSzuttor) [inline src/core/shapes/SimplePore.cpp:38]: Please add docstrings and briefly describe the parameters.

File: src/core/shapes/SimplePore.cpp [hunk 2]
```diff
@@ -0,0 +1,60 @@
+#include "SimplePore.hpp"
+
+#include <cassert>
+
+namespace Shapes {
+std::pair<double, double> SimplePore::dist_half_pore(double r, double z) const {
```
Comment (KaiSzuttor) [inline src/core/shapes/SimplePore.cpp]: Please add docstrings and briefly describe the parameters.

File: src/core/unit_tests/AutoParameter_test.cpp [hunk 3]
```diff
@@ -24,6 +24,27 @@ BOOST_AUTO_TEST_CASE(direct_binding) {
   BOOST_CHECK(i == 42);
 }
 
+BOOST_AUTO_TEST_CASE(method_pointer) {
```
Comment (KaiSzuttor) [inline src/core/unit_tests/AutoParameter_test.cpp:27]: I think this is not related to the PR, but if you want to keep it please describe what the test tests.

File: src/python/espressomd/shapes.py [hunk 4]
```diff
@@ -236,3 +236,25 @@ class Wall(ScriptInterfaceHelper):
...
+    radius: float
+       The radius of the pore.
+    length: float
+       The distance between the planes.
+    smoothing_radius: float
+       Radius of the torus segments
+    axis: array_like
+       Axis of the cylinder and normal of the planes
+    center: array_like
+       Position of the center of the cylinder.
```
Comment (KaiSzuttor) [inline src/python/espressomd/shapes.py:259]: Pls add a newline at the end of the docstring.

File: testsuite/simple_pore.py [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+        for i in range(200):
+            rpos=[i*(box_x / 200.), 0.5 * box_yz, 0.5 * box_yz]
+            s.part.add(id=i, pos=rpos, type=1, v=[1.,1.,1.])
+
+        start_energy = s.analysis.energy()['total']
+        s.integrator.run(1000)
+        end_energy = s.analysis.energy()['total']
+        rel_diff = abs(end_energy-start_energy) / start_energy
+
+        self.assertTrue(rel_diff < 1e-3)
```
Comment (KaiSzuttor) [inline testsuite/simple_pore.py]: self.assertLess(rel_diff, 1e-3)

**Review by KaiSzuttor — CHANGES_REQUESTED** [review]

File: src/core/unit_tests/AutoParameter_test.cpp (hunk 3 above)
Comment (fweik) [reply src/core/unit_tests/AutoParameter_test.cpp:27]: Those are used in the SciptInterface of SimplePore.

**Review by KaiSzuttor — APPROVED** [review]

**Pull request merged**